    1: import numpy as np
    2: import pytest
    3: 
    4: from pandas import (
    5:     MultiIndex,
    6:     Series,
    7: )
    8: import pandas._testing as tm
    9: 
   10: 
   11: class TestRepeat:
   12:     def test_repeat(self):
   13:         ser = Series(np.random.default_rng(2).standard_normal(3), index=["a", "b", "c"])
   14: 
   15:         reps = ser.repeat(5)
   16:         exp = Series(ser.values.repeat(5), index=ser.index.values.repeat(5))
   17:         tm.assert_series_equal(reps, exp)
   18: 
   19:         to_rep = [2, 3, 4]
   20:         reps = ser.repeat(to_rep)
   21:         exp = Series(ser.values.repeat(to_rep), index=ser.index.values.repeat(to_rep))
   22:         tm.assert_series_equal(reps, exp)
   23: 
   24:     def test_numpy_repeat(self):
   25:         ser = Series(np.arange(3), name="x")
   26:         expected = Series(
   27:             ser.values.repeat(2), name="x", index=ser.index.values.repeat(2)
   28:         )
   29:         tm.assert_series_equal(np.repeat(ser, 2), expected)
   30: 
   31:         msg = "the 'axis' parameter is not supported"
   32:         with pytest.raises(ValueError, match=msg):
   33:             np.repeat(ser, 2, axis=0)
   34: 
   35:     def test_repeat_with_multiindex(self):
   36:         # GH#9361, fixed by  GH#7891
   37:         m_idx = MultiIndex.from_tuples([(1, 2), (3, 4), (5, 6), (7, 8)])
   38:         data = ["a", "b", "c", "d"]
   39:         m_df = Series(data, index=m_idx)
   40:         assert m_df.repeat(3).shape == (3 * len(data),)
